src/Microsoft.Health.Fhir.SqlServer/Features/Search/Expressions/Visitors/QueryGenerators/SqlQueryGenerator.cs

```diff
@@ -844,6 +856,16 @@ private void AppendHistoryClause(in IndentedStringBuilder.DelimitedScope delimit
             }
         }
 
+        private void AppendSearchParameterHashClause(in IndentedStringBuilder.DelimitedScope delimited, string tableAlias = null)
+        {
+            if (_isReindexSearch)
+            {
+                delimited.BeginDelimitedElement();
+
+                StringBuilder.Append(VLatest.Resource.SearchParamHash, tableAlias).Append(" != ").Append(Parameters.AddParameter(_searchParameterHash));
```

Comment:
>_searchParameterHash [](start = 129, length = 20)

Can you be null, and would `AddParameter` correctly handle null here?

Reply:
Yes you're right - thanks for catching this, added a null check in 5e584d8.

Here is an example of the updated query:

```
SELECT DISTINCT TOP (101) r.ResourceTypeId, r.ResourceId, r.Version, r.IsDeleted, r.ResourceSurrogateId, r.RequestMethod, CAST(1 AS bit) AS IsMatch, CAST(0 AS bit) AS IsPartial, r.IsRawResourceMetaSet, r.SearchParamHash, r.RawResource
FROM dbo.Resource r WITH(INDEX(IX_Resource_ResourceTypeId_ResourceSurrgateId))
WHERE ResourceTypeId = 81
    AND IsHistory = 0
    AND IsDeleted = 0
    AND (SearchParamHash != '4A2D391F6FCCDF97C5FF67C6B79DCABED44A521FE40F17FA9F0F44DEA6435CBD' OR SearchParamHash IS NULL)
ORDER BY r.ResourceSurrogateId ASC
OPTION(RECOMPILE)
```

Reply:
My cosmos db emulator returns true for this ``SELECT NULL != 'hash'`, but I haven't looked into documentation regarding null comparison so maybe I'm raising false hope.

(All I'm saying, maybe that OR addition is not necessary, but I'm not sure, also my original comment was somewhat generic, I see object passing, I'm curious how it would handle null, I didn't imply something is wrong here, sorry if I confused you.)